src/test/java/se/sundsvall/messaging/integration/db/MessageRepositoryTest.java

```diff
@@ -0,0 +1,96 @@
+package se.sundsvall.messaging.integration.db;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace.NONE;
+import static se.sundsvall.messaging.model.MessageStatus.PENDING;
+import static se.sundsvall.messaging.model.MessageStatus.SENT;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.jdbc.Sql;
+
+import se.sundsvall.messaging.test.annotation.UnitTest;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = NONE)
+@UnitTest
+@Sql(scripts = {
+	"/db/scripts/truncate.sql",
+	"/db/scripts/testdata-ut.sql"
+})
+class MessageRepositoryTest {
+
+	private static final String NON_EXISTING_ID = "d62134e2-f652-42a1-be43-72f620a1a358";
+
+	@Autowired
+	private MessageRepository messageRepository;
+
+	@Test
+	void findByDeliveryId() {
+		// Arrange
+		final var deliveryId = "0cc1cd5b-1196-49e9-9dad-11cc0da77e3d";
+
+		// Act and assert
+		assertThat(messageRepository.findByDeliveryId(deliveryId)).isPresent().map(entity -> {
```

Comment:
Should probably be using .satisfies() instead of map here.

Reply:
See comment regarding Optional above